lib/experimental/html/class-wp-html-attribute-token.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * HTML Walker: Attribute token structure class.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ * @since 6.1.0
+ */
+
+/**
+ * Data structure for the attribute token that allows to drastically improve performance.
+ *
+ * @since 6.1.0
+ *
+ * @see WP_HTML_Walker
+ */
+class WP_HTML_Attribute_Token {
+	/**
+	 * Tag name.
+	 *
+	 * @since 6.1.0
+	 * @var string
+	 */
+	public $name;
+
+	/**
+	 * Attribute value.
+	 *
+	 * @since 6.1.0
+	 * @var string
+	 */
+	public $value;
+
+	/**
+	 * Start position of the token.
```

Comment:
I picked the word `offset` across this code to refer to the idea of index/position/what have you – it would be nice to keep the terminology consistent across the board. Also, this could be a bit more specific – it's the string offset of the attribute name.